include/types.h

```diff
@@ -485,8 +495,37 @@ struct retained_format_field {
 
     retained_format_field(std::vector<std::string> orig_names_, std::string name_, RetainedFieldType type_,
         FieldCombinationMethod combi_method_, RetainedFieldNumber number_, int count_=0, bool default_to_zero_=false)
-        : orig_names(orig_names_), name(name_), type(type_), number(number_), default_to_zero(default_to_zero_), count(count_), combi_method(combi_method_) {}
+        : orig_names(orig_names_), name(name_), type(type_), number(number_), default_to_zero(default_to_zero_), count(count_), combi_method(combi_method_) {
+        // Keep the names in sorted order, so that the comparison operator
+        // will compare orig_name vectors element-wise.
+        std::sort(orig_names.begin(), orig_names.end());
+    }
+
+//    bool operator<(const retained_format_field& rhs) const noexcept {
+//        return name < rhs.name;
+//    }
+
+    bool operator==(const retained_format_field& rhs) const noexcept {
+        // compare the orig_names vector, they are sorted.
+        if (orig_names.size() != rhs.orig_names.size())
+            return false;
+        std::vector<std::string>::const_iterator it, itrhs;
+        for (it = orig_names.begin(), itrhs = rhs.orig_names.begin();
+             it != orig_names.end() && itrhs != rhs.orig_names.end();
+             ++it, ++itrhs) {
+            if ((*it) != (*itrhs)) return false;
+        }
 
+        return name == rhs.name &&
+            description == rhs.description &&
+            type == rhs.type &&
+            number == rhs.number &&
+            default_to_zero == rhs.default_to_zero &&
+            count == rhs.count &&
+            combi_method == rhs.combi_method;
+    }
```

Comment:
I think the only use of the new `{genotyper,unifier}_config:operator==` you've defined is to check the round-trip equality in the unit tests. Is that correct? If so, could you perform the equality check at the YAML text level and avoid the need for these carefully coded operators?